MonkeyWrench.Web.UI/ViewLane.aspx.cs

```
@@ -29,11 +30,58 @@
 public partial class ViewLane : System.Web.UI.Page
 {
 	GetViewLaneDataResponse response;
+	static Dictionary<string, string> protectedBranches = new Dictionary<string, string>();
 
 	private new Master Master
 	{
 		get { return base.Master as Master; }
 	}
+	string ParseRepo(string repo)
+	{
+		var regex  = new Regex(@"github\.com[:/](.*)$");
+		var result = regex.Match(repo).Groups[1].Value;
+		return result.Replace(".git", "");
+	}
+
+	string ParseBranch(string branch)
+	{
+		return branch.Split('/').Last();
+	}
+
+	bool IsBranchProtected(DBLane lane)
+	{
+		var repo = lane.repository;
+		var branch = lane.max_revision;
+
+		if (!repo.Contains("github")) return false;
+
+		var key = repo + ":" + branch;
+		var token = Session["github_token"];
+
+		var url = "https://api.github.com/repos/" + repo + "/branches/" + branch + "/protection";
+
+		var client = WebRequest.Create(url) as HttpWebRequest;
+		client.Accept = "application/vnd.github.loki-preview+json";
+		client.ContentType = "application/json";
+		client.Method = WebRequestMethods.Http.Get;
+		client.PreAuthenticate = true;
+		client.UserAgent = "app";
+
+		client.Headers.Add("Authorization", "token " + token);
+
+		if (protectedBranches.ContainsKey(key)) client.Headers.Add("If-None-Match", protectedBranches[key]);
+
+		try {
+			var resp = client.GetResponse() as HttpWebResponse;
+			if (resp.Headers.AllKeys.Contains("Etag"))
+				protectedBranches.Add(key, resp.Headers["Etag"]);
+
+			return resp.StatusCode == HttpStatusCode.OK || resp.StatusCode == HttpStatusCode.NotModified;
+
+		} catch (WebException) {
+			return false;

```

Might want to log a failure, instead of silently swallowing it.